pkg/source/adapter/adapter.go

```diff
@@ -303,3 +308,34 @@ func (a *Adapter) InitOffsets(session sarama.ConsumerGroupSession) error {
 	// At this stage the KafkaSource instance is considered Ready (TODO: update KafkaSource status)
 	return nil
 }
+
+// Default retry configuration, 5 retries, exponential backoff with 50ms delay
+func defaultRetryConfig() *kncloudevents.RetryConfig {
+	retryConfig := kncloudevents.NoRetries()
+	retryConfig.CheckRetry = func(ctx context.Context, response *nethttp.Response, err error) (bool, error) {
+		// See Knative Eventing spec (WIP) for when to retry
+		// Implement: https://github.com/knative/specs/blob/218f9655c23ca6a35280afc638b2e3c0a5e83f63/specs/eventing/data-plane.md#event-acknowledgement-and-delivery-retry
+
+		// Handle the special cases first
+
+		// do not retry on context.Canceled or context.DeadlineExceeded
+		if ctx.Err() != nil {
+			return false, ctx.Err()
+		}
+
+		if err == nil {
+			if response.StatusCode == http.StatusNotFound || response.StatusCode == http.StatusConflict {
+				return true, nil
+			}
+
+		}
+
+		// Fallback to the default retry policy capturing all other cases.
+		return retryablehttp.DefaultRetryPolicy(ctx, response, err)
+	}
+	retryConfig.RetryMax = 5
+	retryConfig.Backoff = func(attemptNum int, resp *nethttp.Response) time.Duration {
+		return retryablehttp.DefaultBackoff(50*time.Millisecond, 2*time.Second, attemptNum, resp)
```

Comment:
that is easier to read, but adds a new dependency. 

in #675 I can "borrowed" existing code from eventing, rather than adding a new dep :shrug:

Reply:
Checking on the method:

https://github.com/hashicorp/go-retryablehttp/blob/master/client.go#L469-L475

I see it does recognize the `Retry-After` header, which we go some little push back here:
https://github.com/knative/eventing/pull/5285/files

/cc @slinkydeveloper @travis-minke-sap

Reply:
I am personally happy to see this _landing_ (the support for `retry-after`), I just point to the comments of spec breaking changes on the other PR

Reply:
Also - the `DefaultBackoff()` functionality from that library does NOT support BOTH mechanisms (number of seconds, date) as does the POC PR I had created.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After

Reply:
interesting. Good to know, thanks @travis-minke-sap